File: src/java/Generator.java
package java;

import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.net.URI;
import java.Messages.SynapSysRequest;

public class Generator {
    private final String provider;
    private final String model;
    private final String query;
    private final String apiKey;
    private final String synapsysKey;
    private final String serviceEndpoint;
    private final HttpClient httpClient;

    public Generator(String provider, String model, String query, String apiKey, String synapsysKey) {
        this(provider, model, query, apiKey, synapsysKey, "http://localhost:8080/api/generate");
    }

    public Generator(String provider, String model, String query, String apiKey, String synapsysKey,
            String serviceEndpoint) {
        this.provider = provider;
        this.model = model;
        this.query = query;
        this.apiKey = apiKey;
        this.synapsysKey = synapsysKey;
        this.serviceEndpoint = serviceEndpoint;
        this.httpClient = HttpClient.newHttpClient();
    }

    public String generate() {
        try {
            SynapSysRequest request = new SynapSysRequest(provider, model, query, apiKey);
            String jsonBody = request.serialize();

            HttpRequest httpRequest = HttpRequest.newBuilder()
                    .uri(URI.create(serviceEndpoint))
                    .header("Content-Type", "application/json")
                    .header("Authorization", "Bearer " + synapsysKey)
                    .POST(HttpRequest.BodyPublishers.ofString(jsonBody))
                    .build();

            HttpResponse<String> response = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
            return response.body();
        } catch (Exception e) {
            e.printStackTrace();
            return "{\"error\":\"" + escapeJson(e.getMessage()) + "\"}";
        }
    }

    private static String escapeJson(String s) {
        if (s == null)
            return "";
        return s.replace("\\", "\\\\").replace("\"", "\\\"");
    }
}

File: src/python/Generator.py
import json
from urllib import request
from src.python.Messages.SynapSysRequest import SynapSysRequest

class Generator:
    def __init__(self, provider, model, query, api_key, synapsys_key, service_endpoint=None):
        self.provider = provider
        self.model = model
        self.query = query
        self.api_key = api_key
        self.synapsys_key = synapsys_key
        self.service_endpoint = service_endpoint or "http://localhost:8080/api/generate"
    
    def generate(self):
        try:
            synapsys_request = SynapSysRequest(
                provider=self.provider,
                model=self.model,
                query=self.query,
                apiKey=self.api_key
            )
            
            json_body = synapsys_request.serialize()
            
            http_request = request.Request(
                self.service_endpoint,
                data=json_body,
                method="POST",
                headers={
                    "Content-Type": "application/json",
                    "Authorization": f"Bearer {self.synapsys_key}"
                }
            )
            
            with request.urlopen(http_request, timeout=30) as response:
                return response.read().decode('utf-8')
        except Exception as e:
            return json.dumps({"error": str(e)})